app/models/live_result.rb

```diff
@@ -88,8 +91,12 @@ def notify_users
     end
 
     def recompute_advancing
+      has_linked_round = round.linked_round.present?
+      advancement_determining_column = has_linked_round ? 'global_pos' : 'local_pos'
```

Comment:
In case of non-linked rounds, `global_pos` should exactly equal `local_pos`. So simply always picking `global_pos` should be fine